### ISSUE TITLE
Feature request: support altering charset and collation

### ISSUE THREAD

**Issue by sonots** [k0kubun/sqldef#60]
This is a feature request to support altering charset and collation although it is already stated at https://github.com/k0kubun/sqldef/issues/47 as

> Parse column's collate (and maybe charset as well) and support altering them.

Also, it looks charset and collation are not reflected on adding a new column. At least, I am happy if charset and collation on adding a new column is supported.

**Comment by k0kubun** [comment]
> At least, I am happy if charset and collation on adding a new column is supported.

Could you provide a set of "current schema" + "input schema" + "expected DDL"? I'll use it as a test case when implementing the feature.

**Comment by sonots** [comment]
Sure. My team member takes care of it. @hiroyuki-kamei

**Comment by hiroyuki-kamei** [comment]
Thank you for checking immediately.
This is a set.

current schema
```
+-------+------------------------+
| Table | Create Table          |
+-------+------------------------+
| table | CREATE TABLE `table` (
  `id` int(11) NOT NULL,
  `member_id` varchar(254) COLLATE utf8mb4_bin DEFAULT NULL,
  `password` char(128) COLLATE utf8mb4_bin DEFAULT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `member_id` (`member_id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+------------------------+
```

input schema
```
CREATE TABLE `table` (
  `id` INT NOT NULL PRIMARY KEY,
  `member_id` VARCHAR(254) CHARACTER SET latin1 COLLATE latin1_bin UNIQUE,
  `password` CHAR(128) CHARACTER SET latin1 COLLATE latin1_bin
);
```

expected DDL
```
ALTER TABLE members MODIFY `member_id` VARCHAR(254) CHARACTER SET latin1 COLLATE latin1_bin;
ALTER TABLE members MODIFY `password` CHAR(128) CHARACTER SET latin1 COLLATE latin1_bin;
```

The result of executing mysqldef is "Nothing is modified"

**Comment by k0kubun** [comment]
I'm releasing a fix as v0.5.19. Could you try it once [the release](https://travis-ci.org/github/k0kubun/sqldef/builds/705031024) finishes?

**Comment by hiroyuki-kamei** [comment]
@k0kubun 
Thank you for the correction.

However, multiple UNIQUE KEYs are registered.

expected DDL is
```
ALTER TABLE test_table CHANGE COLUMN member_id member_id varchar(254) CHARACTER SET latin1 COLLATE latin1_bin;
ALTER TABLE test_table CHANGE COLUMN password password char(128) CHARACTER SET latin1 COLLATE latin1_bin;
```

but this will be executed.
```
ALTER TABLE test_table CHANGE COLUMN member_id member_id varchar(254) CHARACTER SET latin1 COLLATE latin1_bin UNIQUE;
ALTER TABLE test_table CHANGE COLUMN password password char(128) CHARACTER SET latin1 COLLATE latin1_bin;
```

If the UNIQUE key exists, can it be removed from the created ALTER statement?

---

This is a reproduction case

[1] input schema
Default CHARSET is utf8mb4 and collation is utf8mb4_bin

```
CREATE TABLE `test_table` (
  `id` INT NOT NULL PRIMARY KEY,
  `member_id` VARCHAR(254) UNIQUE,
  `password` CHAR(128)
);
```
The following table will be created
```
test_table | CREATE TABLE `test_table` (
  `id` int(11) NOT NULL,
  `member_id` varchar(254) COLLATE utf8mb4_bin DEFAULT NULL,
  `password` char(128) COLLATE utf8mb4_bin DEFAULT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `member_id` (`member_id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
```

[2] modify input schema to change CHARSET and COLLATE

```
CREATE TABLE `test_table` (
  `id` INT NOT NULL PRIMARY KEY,
  `member_id` VARCHAR(254) CHARACTER SET latin1 COLLATE latin1_bin UNIQUE,
  `password` CHAR(128) CHARACTER SET latin1 COLLATE latin1_bin
);
```
The following ALTER is executed
```
--Apply--
ALTER TABLE test_table CHANGE COLUMN member_id member_id varchar(254) CHARACTER SET latin1 COLLATE latin1_bin UNIQUE;
ALTER TABLE test_table CHANGE COLUMN password password char(128) CHARACTER SET latin1 COLLATE latin1_bin;
```

[3] Multiple UNIQUE KEYs are registered
```
test_table | CREATE TABLE `test_table` (
  `id` int(11) NOT NULL,
  `member_id` varchar(254) CHARACTER SET latin1 COLLATE latin1_bin DEFAULT NULL,
  `password` char(128) CHARACTER SET latin1 COLLATE latin1_bin DEFAULT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `member_id` (`member_id`),
  UNIQUE KEY `member_id_2` (`member_id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
```

**Comment by k0kubun** [comment]
Oh, that was not intentional. Let me take a look at that too.

**Comment by k0kubun** [comment]
I think I fixed the other issue in v0.6.2. Please try the version once [the release](https://travis-ci.org/github/k0kubun/sqldef/builds/705293339) is completed.

**Comment by hiroyuki-kamei** [comment]
The expected DDL was executed in v0.6.2.
Thanks a lot for the fix!
